core/src/main/java/hudson/ExtensionFinder.java

```diff
@@ -486,12 +486,6 @@ private void resolve(Class<?> c, Set<Class<?>> encountered) {
                     return;
                 }
                 try {
-                    ClassLoader ecl = c.getClassLoader();
-                    if (ecl != null) { // Not bootstrap classloader
-                        Method m = ClassLoader.class.getDeclaredMethod("resolveClass", Class.class);
-                        m.setAccessible(true);
-                        m.invoke(ecl, c);
```

Comment:
this call seems to just check if the class is null?

![image](https://user-images.githubusercontent.com/21194782/102010267-00bed700-3d35-11eb-841e-bfb137efcbdd.png)

If I add a null check IDE tell's me that it's impossible for it to be null at this point.

Removing this check doesn't break any tests

Reply:
Not sure what version of Java you are looking at. In Java 8

```java
protected final void resolveClass(Class<?> c) {
    resolveClass0(c);
}

private native void resolveClass0(Class<?> c);
```

Reply:
Java 11 Azul

Reply:
Maybe this became obsolete somehow in 11, but still appears to do something real in 8, so suggest reverting.

Reply:
The code that called this is removed in java 9, https://bugs.openjdk.java.net/browse/JDK-8057777

I'm not sure if it was doing something in 8 or not, the code was removed because it wasn't doing anything but it doesn't say why, it couldn't been something in 9 that meant it wasn't used any more.

Reply:
I can't seem to see where the definition of resolveClass0 lies to confirm no side-effects. It seems difficult to confirm that removing this is safe.

> it couldn't been something in 9 that meant it wasn't used any more.

How did you arrive at this conclusion?

Even if that is the case there is a possibility it was removed in a jdk8 update but which update exactly, there might be breakages on systems with older versions of jdk8 possibly.

Reply:
I read the code, but the openjdk repo is only tagged up to v120 or so of java 8.

the code definitely does nothing on 9.

If people think it’s still worth keeping I can re-add it but only for java 8

Reply:
In http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/ `jdk/src/share/native/java/lang/ClassLoader.c` as of `jdk8u282-b07` has

```cpp
JNIEXPORT void JNICALL
Java_java_lang_ClassLoader_resolveClass0(JNIEnv *env, jobject this,
                                         jclass cls)
{
    if (cls == NULL) {
        JNU_ThrowNullPointerException(env, 0);
        return;
    }

    JVM_ResolveClass(env, cls);
}
```

which is defined in `hotspot/src/share/vm/prims/jvm.cpp` as

```cpp
JVM_ENTRY(void, JVM_ResolveClass(JNIEnv* env, jclass cls))
  JVMWrapper("JVM_ResolveClass");
  if (PrintJVMWarnings) warning("JVM_ResolveClass not implemented");
JVM_END
```

and this code does not appear to have been edited since at least 2007! I suppose the Javadoc remains because another JVM could in principle implement this, but seeing as this was just some old heuristic to work around alleged Guice problems, I concur that this is safe to remove.